src/cli/config-flags.ts

```diff
@@ -53,8 +106,71 @@ export const STRING_CLI_ARGS = [
   'emulate',
   'root',
   'screenshotConnector',
+
+  // JEST CLI ARGS
+  'cacheDirectory',
+  'changedSince',
+  'collectCoverageFrom',
+  // 'config', Stencil already supports this argument
+  'coverageDirectory',
+  'coverageThreshold',
+  'env',
+  'filter',
+  'globalSetup',
+  'globalTeardown',
+  'globals',
+  'haste',
+  'moduleNameMapper',
+  'notifyMode',
+  'outputFile',
+  'preset',
+  'prettierPath',
+  'resolver',
+  'rootDir',
+  'runner',
+  'testEnvironment',
+  'testEnvironmentOptions',
+  'testFailureExitCode',
+  'testNamePattern',
+  'testResultsProcessor',
+  'testRunner',
+  'testSequencer',
+  'testURL',
+  'timers',
+  'transform',
+
+  // ARRAY ARGS
+  'collectCoverageOnlyFrom',
+  'coveragePathIgnorePatterns',
+  'coverageReporters',
+  'moduleDirectories',
+  'moduleFileExtensions',
+  'modulePathIgnorePatterns',
+  'modulePaths',
+  'projects',
+  'reporters',
+  'roots',
+  'selectProjects',
+  'setupFiles',
+  'setupFilesAfterEnv',
+  'snapshotSerializers',
+  'testMatch',
+  'testPathIgnorePatterns',
+  'testPathPattern',
+  'testRegex',
+  'transformIgnorePatterns',
+  'unmockedModulePathPatterns',
+  'watchPathIgnorePatterns',
 ] as const;
 
+/**
+ * All the CLI arguments which may have string or number values
+ *
+ * `maxWorkers` is an argument which is used both by Stencil _and_ by Jest,
+ * which means that we need to support parsing both string and number values.
+ */
+export const STRING_NUMBER_CLI_ARGS = ['maxWorkers'] as const;
```

Comment:
`maxWorkers` [can according to the Jest docs](https://jestjs.io/docs/27.x/cli#--maxworkersnumstring) be either a string or a number, so we need to introduce a new argument type to support that.